python/pyarrow/tests/parquet/test_parquet_encryption.py

```diff
@@ -0,0 +1,301 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import pyarrow as pa
+import pytest
+
+try:
+    import pyarrow.parquet as pq
+except ImportError:
+    pq = None
+
+import base64
+from Cryptodome.Cipher import AES
+from collections import OrderedDict
+from datetime import timedelta
+
+DATA_TABLE = pa.Table.from_pydict(
+    OrderedDict([
+        ('a', pa.array([1, 2, 3])),
+        ('b', pa.array(['a', 'b', 'c'])),
+        ('c', pa.array(['x', 'y', 'z']))
+    ])
+)
+PARQUET_NAME = 'encrypted_table.in_mem.parquet'
+FOOTER_KEY = "0123456789ABCDEF"
```

Comment:
Might be a little nicer to have `FOOTER_KEY` and `COL_KEY` be bytes to begin with: simplifies the example KMS client slightly, and removes the mental question of "why aren't the keys bytes".

Reply:
Thank you, @itamarst . We've changed to generated keys. The KMS client still gets these as strings, since the test is also checking usage of `KmsConnectionConfig` with `custom_kms_conf `, which is a dictionary of strings. 
Anyway, this KMS client is not to be used as an example, but we are going to add a Vault KMS client example.